templates/submission/row.jade

```diff
@@ -3,7 +3,7 @@
 - load i18n
 
 if profile_id
-    - var can_view = submission.problem_id in authored_problem_ids or profile_id == submission.user_id or perms['judge']['change_submission'] or submission.problem_id in completed_problem_ids and submission.problem.is_public
+    - var can_view = profile_id in submission.problem.editor_ids or profile_id == submission.user_id or perms['judge']['change_submission'] or submission.problem_id in completed_problem_ids and (submission.problem.is_public or profile_id in submission.problem.tester_ids)
```

Comment:
What the hell. Do not do this. It is obvious why `authored_problem_ids` was used. You ought to change this to `editable_problem_ids` and implement it that way.